src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java

```diff
@@ -549,4 +559,223 @@ public AdherenceStatistics getAdherenceStatistics(String appId, String studyId,
         }
         return reportDao.getAdherenceStatistics(appId, studyId, adherenceThreshold);
     }
+    
+    public DetailedAdherenceReport getDetailedAdherenceReportForParticipant(String appId, String studyId, Account account) {
+        checkNotNull(appId);
+        checkNotNull(studyId);
+        checkNotNull(account);
+        
+        // Collect adherence state.
+        DateTime createdOn = getDateTime();
+        String zoneId = studyService.getZoneId(appId, studyId, account.getClientTimeZone());
+        
+        AdherenceState.Builder builder = new AdherenceState.Builder();
+        builder.withNow(createdOn);
+        builder.withClientTimeZone(zoneId);
+        
+        Study study = studyService.getStudy(appId, studyId, true);
+        if (study.getScheduleGuid() == null) {
+            throw new EntityNotFoundException(Schedule2.class);
+        }
+        Schedule2 schedule = scheduleService.getScheduleForStudy(appId, studyId)
+                .orElseThrow(() -> new EntityNotFoundException(Schedule2.class));
+        List<TimelineMetadata> metadata = scheduleService.getScheduleMetadata(study.getScheduleGuid());
+        
+        Timeline timeline = scheduleService.getTimelineForSchedule(appId, schedule.getGuid());
+        
+        List<StudyActivityEvent> events = studyActivityEventService.getRecentStudyActivityEvents(
+                appId, studyId, account.getId()).getItems();
+        
+        List<AdherenceRecord> adherenceRecords = getAdherenceRecords(appId, new AdherenceRecordsSearch.Builder()
```

Comment:
This could get over that limit, especially because it includes assessment adherence records. The other reports use only sessions so they haven't hit a need to go over a page. I updated it to collect using pagination.

I used the max page size constant though, which is set to 100. Do you think I should increase it to 250 or 500?